pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -295,5 +299,87 @@ func manageOpenShiftControllerManagerDeployment_v311_00_to_latest(client appscli
 	}
 	required.Annotations[util.VersionAnnotation] = os.Getenv("RELEASE_VERSION")
 
+	proxyCfg, err := proxyLister.Get("cluster")
```

Comment:
We should have error handling logic here:

1. If `err` is a `NotFound` error, consider the proxy env vars unset, proceed with logic below.
2. If `err` something else, bubble out (should trigger a re-queue).

Reply:
So I definitely will update with add a recoreder event and bubbling out / retriggering more immediately - thanks.

On the not found piece, I did some experiments, and
a) the instance is there on a standard install
b) you are not allowed to delete it, even with system:admin, from an install:

```
gmontero ~/go/src/github.com/openshift/cluster-openshift-controller-manager-operator  (ocm-proxy-envs)$ oc get proxies.config.openshift.io
NAME      AGE
cluster   83m
gmontero ~/go/src/github.com/openshift/cluster-openshift-controller-manager-operator  (ocm-proxy-envs)$ oc delete  proxies.config.openshift.io cluster 
Error from server (Forbidden): proxies.config.openshift.io "cluster" is forbidden: deleting required proxies.config.openshift.io resource, named cluster, is not allowed
gmontero ~/go/src/github.com/openshift/cluster-openshift-controller-manager-operator  (ocm-proxy-envs)$ 
```

I'm inclined to minimally record an event if we unexpectedly see not found.

I am afraid that returning the errror and retriggering could create a vicious cycle/loop if the cluster is actually broken, so think will proceed as is as noted.

But further opinions of course welcomed.

Reply:
update pushed in separate commit @adambkaplan